src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/tools/monitoring/outputMonitor.ts

```diff
@@ -247,42 +248,54 @@ export class OutputMonitor extends Disposable implements IOutputMonitor {
 		const maxWaitMs = extendedPolling ? PollingConsts.ExtendedPollingMaxDuration : PollingConsts.FirstPollingMaxDuration;
 		const maxInterval = PollingConsts.MaxPollingIntervalDuration;
 		let currentInterval = PollingConsts.MinPollingDuration;
-
-		let lastBufferLength = execution.getOutput().length;
-		let noNewDataCount = 0;
-		let buffer = '';
 		let waited = 0;
+		let noNewDataCount = 0;
+		let dataChanged = false;
+		let currentOutput: string | undefined;
+		let onDataDisposable = Disposable.None;
 
-		while (!token.isCancellationRequested && waited < maxWaitMs) {
-			const waitTime = Math.min(currentInterval, maxWaitMs - waited);
-			await timeout(waitTime, token);
-			waited += waitTime;
-			currentInterval = Math.min(currentInterval * 2, maxInterval);
-
-			buffer = execution.getOutput();
-			const len = buffer.length;
+		try {
+			while (!token.isCancellationRequested && waited < maxWaitMs) {
+				const waitTime = Math.min(currentInterval, maxWaitMs - waited);
+				await timeout(waitTime, token);
+				waited += waitTime;
+				currentInterval = Math.min(currentInterval * 2, maxInterval);
+				if (currentOutput === undefined) {
```

Comment:
this is defined within the loop because if we don't await that timeout, the buffer is empty